### ISSUE TITLE
Browse "continuations"

### ISSUE THREAD

**Issue by richard-jones** [DOAJ/doaj#145]
When we have a system capable of managing continuations, we will also need a way to expose that information through the public UI

**Comment by richard-jones** [comment]
@markmacgillivray - I'm assigning this to you to start with just because it relates to the ToC pages cache review #541 .  Continuations also need to be supported here, which may increase the complexity somewhat, especially if generating the ToCs on the fly.

**Comment by markmacgillivray** [comment]
What is a continuation?

**Comment by richard-jones** [comment]
I can go over this in detail probably on a call at some point, but in outline:

When a journal changes its name, it usually gets a new ISSN.  This means "the same" journal now has a new identifier, and this is a "continuation" of the old version of the journal.

At the moment we have a single Journal object which represents all the versions of the journal, and we have a "history" field in which the old continuations reside (not to be confused with an edit history for the journal record - this history is only "official" changes to the journal in the real world).

We do not yet have any formal front-end for managing continuations, for the time being the data is just there so it is not lost (it came from the previous incarnation of DOAJ), and so that people can search on it.  The first step to formal support is for us to allow the various continuations to be visible on the ToC pages.  In a future project we will need to add the capability to add continuations, but that's not covered here.

**Comment by dommitchell** [comment]
@richard-jones here's an interesting question: should a publisher fill in a reapplication for a journal that has a continuation? A publisher has got in touch and is wondering whether she should just delete the column from the CSV file.

**Comment by richard-jones** [comment]
So, we have asked them to reapply for a journal that has since been replaced by a new one under a new ISSN in the real world, but for which the DOAJ does not yet have the continuation registered?

Answer in several parts:

* They could still fill in the reapplication, as long as they don't change the ISSN.  Not sure if this would be the right thing to do, though
* We can record continuations yet, at least not in a manageable way, so they can't give us the new continuation in a straightforward manner
* They could just apply separately for the new continuation, provided we later commit to being able to reconcile the chain of continuations.  We would also need to temporarily withdraw the old journal from the DOAJ, which would have consequences for ToC and other such.

Basically, we need formalised continuations support for this to work smoothly, everything else would be a hack.

**Comment by dommitchell** [comment]
@richard-jones @markmacgillivray can you review this and make sure that the continuation solution you implement fits these recommendations? http://www.niso.org/publications/rp/rp-16-2013

**Comment by markmacgillivray** [comment]
@richard-jones I do not know if I should be doing anything with this just now or not - the issue is opened with the phrase "when we have a system that handles continuations". Is that the case? What I have done so far to ToC just does what it did before, only more efficiently.

**Comment by richard-jones** [comment]
Yes, we discussed it back near the start of the project - we can go over it
in the meeting on Monday.

We need to lay out the ToC page so that users can see that there's a
continuation (old version of the journal), and can go to a page for that
old instance of the journal and see the volumes and issues associated with
it from those days.  Probably it's another tab/button on the main journal
ToC page.

Should be relatively straightforward to extend the queries you've got for
the new ToCs to work for a specific continuation, so just a matter of
deciding how to present this and constraining the queries to the relevant
ISSNs.



On 29 March 2015 at 16:23, markmacgillivray <notifications@github.com>
wrote:

> @richard-jones <https://github.com/richard-jones> I do not know if I
> should be doing anything with this just now or not - the issue is opened
> with the phrase "when we have a system that handles continuations". Is that
> the case? What I have done so far to ToC just does what it did before, only
> more efficiently.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DOAJ/doaj/issues/145#issuecomment-87427234>.
>



-- 

Richard Jones,

Founder, Cottage Labs
t: @richard_d_jones, @cottagelabs
w: http://cottagelabs.com

**Comment by markmacgillivray** [comment]
Yes, but I mean I'm not doing any page redesign stuff, so I don't know
about what the continuation may look like when it is displayed. Is this
something I shall come back to after you've done your display stuff?

**Comment by richard-jones** [comment]
Yup sure - I'm doing a first pass on the ToC design today, though it isn't
going to change structurally in any major way.  Just the layout of the
actual info.  Will try to have a PR in for tomorrow morning so we can
review.

On 29 March 2015 at 16:45, markmacgillivray <notifications@github.com>
wrote:

> Yes, but I mean I'm not doing any page redesign stuff, so I don't know
> about what the continuation may look like when it is displayed. Is this
> something I shall come back to after you've done your display stuff?
> On 29 Mar 2015 16:31, "Richard Jones" <notifications@github.com> wrote:
>
> > Yes, we discussed it back near the start of the project - we can go over
> it
> > in the meeting on Monday.
> >
> > We need to lay out the ToC page so that users can see that there's a
> > continuation (old version of the journal), and can go to a page for that
> > old instance of the journal and see the volumes and issues associated
> with
> > it from those days. Probably it's another tab/button on the main journal
> > ToC page.
> >
> > Should be relatively straightforward to extend the queries you've got for
> > the new ToCs to work for a specific continuation, so just a matter of
> > deciding how to present this and constraining the queries to the relevant
> > ISSNs.
> >
> >
> >
> > On 29 March 2015 at 16:23, markmacgillivray <notifications@github.com>
> > wrote:
> >
> > > @richard-jones <https://github.com/richard-jones> I do not know if I
> > > should be doing anything with this just now or not - the issue is
> opened
> > > with the phrase "when we have a system that handles continuations". Is
> > that
> > > the case? What I have done so far to ToC just does what it did before,
> > only
> > > more efficiently.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/DOAJ/doaj/issues/145#issuecomment-87427234>.
> > >
> >
> >
> >
> > --
> >
> > Richard Jones,
> >
> > Founder, Cottage Labs
> > t: @richard_d_jones, @cottagelabs
> > w: http://cottagelabs.com
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/DOAJ/doaj/issues/145#issuecomment-87427725>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DOAJ/doaj/issues/145#issuecomment-87430486>.
>



-- 

Richard Jones,

Founder, Cottage Labs
t: @richard_d_jones, @cottagelabs
w: http://cottagelabs.com

**Comment by markmacgillivray** [comment]
Do now so that when hitting toc by uuid we default to the issn of the latest journal version. See journal index history. Should only show info of latest issn. For old issns show a nav above journal/issue so it becomes issn/journal/issue.

The query for the right articles to show currently uses all available issns, so should be limited just to issn of current toc page.

**Comment by dommitchell** [comment]
As mentioned in our last meeting, DOAJ is trying to adhere to NISO's PIE-J (presentation and identification of e-journals) recommendations. The following are pertinent for continuations:

- Associate articles, issues, volumes, and dates with the journal title under which they were originally published. Identify all content from a former title(s) under that journal title and not the current journal title.
- Ensure that all outputs by the publisher or provider (e.g., Table of Contents alerts, information given to link resolvers, and exporting or e-mailing citations or articles) use the journal title and other identifying citation information under which the content was originally published. Users appreciate as full a journal title history as possible to show clear relationships such as previous or later titles.
- Provide a journal title history. Include the full journal title, publication date range, and ISSN for the current title and at least the immediately preceding and/or succeeding titles, as appropriate.
- Show all ISSN for a title and specify the format for each, e.g., ISSN XXXX-XXXX (Print); ISSN YYYY-YYYY (Online); ISSN-L XXXX-XXXX (Linking ISSN).
- Provide both display and search access to the current and all former journal titles to ensure that all titles hosted on a website are available in browse lists and search features.

@richard-jones this means that we must make space for former titles and ISSNs on the journal homepage

**Comment by richard-jones** [comment]
Also, from #585:

* Add a content range or date range for journals with continuations, next to the appropriate ISSN(s), although this might change once I've seen the treatment in #145

**Comment by richard-jones** [comment]
I've done a bit of work around continuations this week which impacts on this task:

I've modified the ToC page so that it takes 3 new arguments for the template:

* bibjson - the bibjson record which forms the core of the ToC page
* future - a list of bibjson records (most recent first) of future versions of the journal
* past - a list of bibjson records (most recent first) of previous versions of the journal

In the default case of looking at the most recent version, bibjson is the current journal's bibjson, future is an empty list and past is all previous versions.

I have not implemented anything that would allow anything other than the default case yet, but when that is done then bibjson should contain the continuation being viewed, future will contain all continuations (including the most recent) that came after it, and past will contain all those that came before it.

On the Journal model object I have added a couple of features:

* Journal.history() now guarantees the order of the history by inspecting the annotations on the entries for the correct replaced/isreplacedby tags
* Journal.get_history_around() takes an ISSN and gives you the future and past bibjson records (correctly ordered) around a continuation with the given ISSN (result does not include the bibjson record for the ISSN itself only those around it).

In the /toc endpoint, then, you can use Journal.get_history_around() to get the future and past lists to send to the template easily.

Off the back of this, I propose the following canonical URL structure for the ToCs to support the continuations

* /[journal id] - link to the current version of the journal
* /[journal id]/[vol]/[issue] - link to the current version of the journal's volume or issue
* /[journal id]/[issn] - link to the specific continuation of the journal, most recent vol/issue
* /[journal id]/[issn]/[vol]/[issue] - link to the specific continuation at the given vol/issue

Then we can have non-canonical urls which redirect, such as:

* /[issn] - redirect to /[journal id]/[issn]
* /[issn]/[vol]/[issue] - redirect to /[journal id]/[issn]/[vol]/[issue]

**Comment by richard-jones** [comment]
Ok, I have implemented the ability to view previous versions of journals, and to navigate up and down the chain of continuations.

I have implemented the following url routing

* /[journal id] - view current journal
* /[issn] - view journal by specific issn, which might be a previous version
* /[issn | identifier]/[vol]/[issue] - view specific volume/issue for the current version or given continuation

@emanuil-tolev - assigning this to you now, as I'm not sure how it intersects with your URL routing changes that you want to make.

**Comment by richard-jones** [comment]
@emanuil-tolev if you have a PR for this that you want me to review before our roll-out today, just send it on through

**Comment by dommitchell** [comment]
I'd like to show an example of a continuation to the public. Can you find me one easily?

**Comment by dommitchell** [comment]
There are 223 journals with continuations in DOAJ currently. 

Plant Knowledge Journal

**Comment by dommitchell** [comment]
@richard-jones can you get me an example of a continuation that has article metadata too?

**Comment by Steven-Eardley** [comment]
This is one I found for testing OpenURL:

https://doaj.org/openurl?url_ver=Z39.88-2004&url_ctx_fmt=info:ofi/fmt:kev:mtx:ctx&rft_val_fmt=info:ofi/fmt:kev:mtx:journal&rft.genre=journal&rft.issn=0366-1644&rft.volume=47&rft.issue=1

Unfortunately, a bug (#696) is preventing it from appearing correctly. It should show the continuation's toc page with the volume and issue highlighted, but it's being incorrectly rerouted to the current ISSN, which doesn't have that vol/iss. Hopefully that'll be fixed soon.

**Comment by dommitchell** [comment]
This is great! Thanks Steve.

**Comment by emanuil-tolev** [comment]
This works now, confirmed in #696 by @dommitchell .
